libnavui-dropin/src/test/java/com/mapbox/navigation/dropin/lifecycle/UIViewModelTest.kt

```diff
@@ -26,42 +49,60 @@ class UIViewModelTest {
     fun `state will change change when view model is attached`() = runBlockingTest {
         val viewModel = AdderViewModel(10)
 
-        viewModel.onAttached(mockk())
+        val mapboxNavigation = mockMapboxNavigation()
+        viewModel.onAttached(mapboxNavigation)
         viewModel.invoke(3)
-        viewModel.onDetached(mockk())
+        viewModel.onDetached(mapboxNavigation)
 
         assertEquals(13, viewModel.state.value)
     }
 
     @Test
-    fun `state does not change when view model is detached`() = runBlockingTest {
```

Comment:
This test was describing a behavior created. I was concerned about this before. There was also no reason to say that it should not update state when detached.

Reality showing that we should be able to update state even after detached. But now it is when MapboxNavigation is non existent that the state does not update